shell/platform/tizen/external_texture_surface_gl.cc

```diff
@@ -19,102 +21,62 @@ EVAS_GL_GLOBAL_GLES3_DECLARE();
 #include <GLES3/gl32.h>
 #endif
 
-#include <atomic>
-#include <iostream>
+#include <tbm_surface.h>
 
 #include "flutter/shell/platform/tizen/tizen_log.h"
 
-struct ExternalTextureGLState {
-  GLuint gl_texture;
-};
-
-static std::atomic_long nextTextureId = {1};
-
-static void MarkTbmSurfaceToUse(void* surface) {
-#ifndef WEARABLE_PROFILE
-  FT_ASSERT(surface);
-  tbm_surface_h tbm_surface = (tbm_surface_h)surface;
-  tbm_surface_internal_ref(tbm_surface);
-#endif
-}
-
-static void UnmarkTbmSurfaceToUse(void* surface) {
-  FT_ASSERT(surface);
-  tbm_surface_h tbm_surface = (tbm_surface_h)surface;
-#ifndef WEARABLE_PROFILE
-  tbm_surface_internal_unref(tbm_surface);
-#else
-  tbm_surface_destroy(tbm_surface);
-#endif
+static void OnCollectTexture(void* textureGL) {
+  auto* weak_texture = (std::weak_ptr<ExternalTexture>*)textureGL;
+  auto strong_texture = weak_texture->lock();
+  delete weak_texture;
+  if (strong_texture) {
+    strong_texture->OnDestruction();
+  }
 }
 
-ExternalTextureGL::ExternalTextureGL()
-    : state_(std::make_unique<ExternalTextureGLState>()),
-      available_tbm_surface_(nullptr),
-      texture_id_(nextTextureId++) {}
+ExternalTextureSurfaceGL::ExternalTextureSurfaceGL(
+    FlutterDesktopGpuBufferTextureCallback texture_callback,
+    FlutterDesktopDestructionCallback destruction_callback,
+    void* user_data)
+    : ExternalTexture(),
+      texture_callback_(texture_callback),
+      destruction_callback_(destruction_callback),
+      user_data_(user_data) {}
 
-ExternalTextureGL::~ExternalTextureGL() {
+ExternalTextureSurfaceGL::~ExternalTextureSurfaceGL() {
   if (state_->gl_texture != 0) {
     glDeleteTextures(1, &state_->gl_texture);
   }
-
-  // If there is a available_tbm_surface_ that is not populated, remove it
-  if (available_tbm_surface_) {
-    UnmarkTbmSurfaceToUse(available_tbm_surface_);
-  }
-
   state_.release();
 }
 
-bool ExternalTextureGL::OnFrameAvailable(tbm_surface_h tbm_surface) {
-  if (!tbm_surface) {
-    FT_LOGE("[texture id:%ld] tbm_surface is null", texture_id_);
+bool ExternalTextureSurfaceGL::PopulateTexture(
+    size_t width,
+    size_t height,
+    FlutterOpenGLTexture* opengl_texture) {
+  const FlutterDesktopGpuBuffer* gpu_buffer =
```

Comment:
Do we release this ``gpu_buffer``?

Reply:
gpu_buffer is created at plugin side，this is determined by the plugin.